src/Compilers/CSharp/Test/Semantic/Semantics/AnonymousFunctionTests.cs

```diff
@@ -0,0 +1,802 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [WorkItem(275, "https://github.com/dotnet/csharplang/issues/275")]
+    [CompilerTrait(CompilerFeature.AnonymousFunctions)]
+    public class AnonymousFunctionTests : CSharpTestBase
+    {
+        public static CSharpCompilation VerifyInPreview(string source, params DiagnosticDescription[] expected)
+            => CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(expected);
+
+        [Fact]
+        public void DisallowInNonPreview()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public static int a;
+
+    public void F()
+    {
+        Func<int> f = static () => a;
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
```

Comment:
>CreateCompilation [](start = 12, length = 17)

explicit language version would be good here (`Regular8`) so that we don't have to update this test again later